llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/StatsRecordingThreadPool.java

```diff
@@ -161,98 +149,44 @@ private void setupMDCFromNDC(final Callable<V> actualCallable) {
-            FileSystem.Statistics.StatisticsData threadFSStats = statistics
-                .getThreadStatistics();
-            List<LlapUtil.StatisticsData> allStatsBefore = LlapUtil
-                .getStatisticsForScheme(scheme, statsBefore);
-            long bytesReadDelta = 0;
-            long bytesWrittenDelta = 0;
-            long readOpsDelta = 0;
-            long largeReadOpsDelta = 0;
-            long writeOpsDelta = 0;
-            // there could be more scheme after execution as execution might be accessing a
-            // different filesystem. So if we don't find a matching scheme before execution we
-            // just use the after execution values directly without computing delta difference
-            if (allStatsBefore != null && !allStatsBefore.isEmpty()) {
-              for (LlapUtil.StatisticsData sb : allStatsBefore) {
-                bytesReadDelta += threadFSStats.getBytesRead() - sb.getBytesRead();
-                bytesWrittenDelta += threadFSStats.getBytesWritten() - sb.getBytesWritten();
-                readOpsDelta += threadFSStats.getReadOps() - sb.getReadOps();
-                largeReadOpsDelta += threadFSStats.getLargeReadOps() - sb.getLargeReadOps();
-                writeOpsDelta += threadFSStats.getWriteOps() - sb.getWriteOps();
-              }
-            } else {
-              bytesReadDelta = threadFSStats.getBytesRead();
-              bytesWrittenDelta = threadFSStats.getBytesWritten();
-              readOpsDelta = threadFSStats.getReadOps();
-              largeReadOpsDelta = threadFSStats.getLargeReadOps();
-              writeOpsDelta = threadFSStats.getWriteOps();
-            }
-            tezCounters.findCounter(scheme, FileSystemCounter.BYTES_READ)
-                .increment(bytesReadDelta);
-            tezCounters.findCounter(scheme, FileSystemCounter.BYTES_WRITTEN)
-                .increment(bytesWrittenDelta);
-            tezCounters.findCounter(scheme, FileSystemCounter.READ_OPS).increment(readOpsDelta);
-            tezCounters.findCounter(scheme, FileSystemCounter.LARGE_READ_OPS)
-                .increment(largeReadOpsDelta);
-            tezCounters.findCounter(scheme, FileSystemCounter.WRITE_OPS)
-                .increment(writeOpsDelta);
+          LlapThreadLocalStatistics currentStats = new LlapThreadLocalStatistics(mxBean);
+          currentStats.subtract(statsBefore).fill(tezCounters);
 
-            if (LOG.isDebugEnabled()) {
-              LOG.debug("Updated stats: instance: {} thread name: {} thread id: {} scheme: {} " +
-                      "bytesRead: {} bytesWritten: {} readOps: {} largeReadOps: {} writeOps: {}",
-                  actualCallable.getClass().getSimpleName(), thread.getName(), thread.getId(),
-                  scheme, bytesReadDelta, bytesWrittenDelta, readOpsDelta, largeReadOpsDelta,
-                  writeOpsDelta);
-            }
+          if (LOG.isDebugEnabled()) {
```

Comment:
Why do we need to check if debug logging is enabled? I mean LOG.debug logs only when it is enabled.

Reply:
this is the usual debug check in cases the string creation might be already expensive, so we don't even want to to it on INFO level

Reply:
Ok.